TasksResearches/task_10.md

```diff
@@ -0,0 +1,33 @@
+## Path Traversal Attacks and Double Encoding:
+
+### Path Traversal Attack
+
+- Path traversal is about sneaking into files you shouldn’t access.
+- A path traversal attacks happens when someone or attacker is able to access files or directories that are not allowed to him to show as an example; the files and directories related to admins. Not all users are allowed to browse them.
+- Simply the attacker can exceed the security limits set to gain access to files or information they have no access to see.
+
+#### To Prevent It:
+- Make sure the file paths users provide are checked carefully.
+- Only allow access to files you know are safe and in the right place.
+
+***
+
+### Double Encoding
+
+- Double encoding is about tricking systems that try to guard against tricky inputs.
+- Double encoding is a technique used by attackers to trick website security systems. When a website tries to protect itself from dangerous inputs, it has a system to check for encoding.
+Double encoding means the attacker encodes their input twice or more to bypass the filters that protect the site.
+
+
+#### To Prevent It
+- Make sure you decode inputs completely before checking them.
+- Use built-in tools and libraries designed to handle encoding safely.
+
+***
+
+### But by using Laravel, it protects against both path Traversal and Double encoding attacks through robust validation techniques, access control tools, and secure handling of inputs.
```

Comment:
Laravel protects us from such attacks, but we can make attacks with our codes so i will illustrate U a code that cause such attack in the session, if we have enough time.. probably we don't😂